rust/protocol/src/protocol.rs

```diff
@@ -358,10 +401,10 @@ impl TryFrom<&[u8]> for PreKeySignalMessage {
         let base_key = PublicKey::deserialize(base_key.as_ref())?;
 
         Ok(PreKeySignalMessage {
-            message_version,
-            registration_id: proto_structure.registration_id.unwrap_or(0),
-            pre_key_id: proto_structure.pre_key_id,
-            signed_pre_key_id,
+            message_version: message_version.try_into()?,
+            registration_id: (proto_structure.registration_id.unwrap_or(0) as u32).into(),
```

Comment:
Why `as u32` here? Shouldn't `proto_structure.registration_id` already have the right type?

Reply:
It turns out that it does!